[AUTOCUT] Distribution Build Failed for k-NN-3.0.0

***Received Error***: **Error building k-NN, retry with: ./build.sh manifests/3.0.0/opensearch-3.0.0.yml --component k-NN**.
                    k-NN failed during the distribution build for version: 3.0.0.
                    Please see build log at https://build.ci.opensearch.org/job/distribution-build-opensearch/9693/display/redirect.
                    The failed build stage will be marked as unstable(!). Please see ./build.sh step for more details

https://build.ci.opensearch.org/blue/rest/organizations/jenkins/pipelines/distribution-build-opensearch/runs/9712/nodes/75/steps/537/log/?start=0

```
[100%] [32m[1mLinking CXX shared library release/opensearchknn_nmslib.dll[0m
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x48): undefined reference to `knn_jni::JNIUtil::Initialize(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x8f): undefined reference to `knn_jni::JNIUtil::Uninitialize(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x10e): undefined reference to `knn_jni::JNIUtil::CatchCppExceptionAndThrowJava(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x190): undefined reference to `knn_jni::JNIUtil::CatchCppExceptionAndThrowJava(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x21d): undefined reference to `knn_jni::JNIUtil::CatchCppExceptionAndThrowJava(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x286): undefined reference to `knn_jni::JNIUtil::CatchCppExceptionAndThrowJava(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.text+0x2e0): undefined reference to `knn_jni::JNIUtil::CatchCppExceptionAndThrowJava(JNIEnv_*)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(org_opensearch_knn_jni_NmslibService.cpp.obj):org_opensearch_knn_jni_NmslibService.cpp:(.rdata$.refptr._ZTVN7knn_jni7JNIUtilE[.refptr._ZTVN7knn_jni7JNIUtilE]+0x0): undefined reference to `vtable for knn_jni::JNIUtil'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.text+0x6c4): undefined reference to `knn_jni::GetJObjectFromMapOrThrow(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, _jobject*, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, _jobject*> > >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.text+0x1190): undefined reference to `knn_jni::GetJObjectFromMapOrThrow(std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, _jobject*, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, _jobject*> > >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni15NEG_DOT_PRODUCTB5cxx11E[.refptr._ZN7knn_jni15NEG_DOT_PRODUCTB5cxx11E]+0x0): undefined reference to `knn_jni::NEG_DOT_PRODUCT[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni13INNER_PRODUCTB5cxx11E[.refptr._ZN7knn_jni13INNER_PRODUCTB5cxx11E]+0x0): undefined reference to `knn_jni::INNER_PRODUCT[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni11COSINESIMILB5cxx11E[.refptr._ZN7knn_jni11COSINESIMILB5cxx11E]+0x0): undefined reference to `knn_jni::COSINESIMIL[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni4LINFB5cxx11E[.refptr._ZN7knn_jni4LINFB5cxx11E]+0x0): undefined reference to `knn_jni::LINF[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni2L1B5cxx11E[.refptr._ZN7knn_jni2L1B5cxx11E]+0x0): undefined reference to `knn_jni::L1[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni2L2B5cxx11E[.refptr._ZN7knn_jni2L2B5cxx11E]+0x0): undefined reference to `knn_jni::L2[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni10SPACE_TYPEB5cxx11E[.refptr._ZN7knn_jni10SPACE_TYPEB5cxx11E]+0x0): undefined reference to `knn_jni::SPACE_TYPE[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni21INDEX_THREAD_QUANTITYB5cxx11E[.refptr._ZN7knn_jni21INDEX_THREAD_QUANTITYB5cxx11E]+0x0): undefined reference to `knn_jni::INDEX_THREAD_QUANTITY[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni8M_NMSLIBB5cxx11E[.refptr._ZN7knn_jni8M_NMSLIBB5cxx11E]+0x0): undefined reference to `knn_jni::M_NMSLIB[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni1MB5cxx11E[.refptr._ZN7knn_jni1MB5cxx11E]+0x0): undefined reference to `knn_jni::M[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni22EF_CONSTRUCTION_NMSLIBB5cxx11E[.refptr._ZN7knn_jni22EF_CONSTRUCTION_NMSLIBB5cxx11E]+0x0): undefined reference to `knn_jni::EF_CONSTRUCTION_NMSLIB[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni15EF_CONSTRUCTIONB5cxx11E[.refptr._ZN7knn_jni15EF_CONSTRUCTIONB5cxx11E]+0x0): undefined reference to `knn_jni::EF_CONSTRUCTION[abi:cxx11]'
C:/Users/ContainerAdministrator/scoop/apps/mingw/12.2.0-rt_v10-rev1/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/opensearchknn_nmslib.dir/objects.a(nmslib_wrapper.cpp.obj):nmslib_wrapper:(.rdata$.refptr._ZN7knn_jni10PARAMETERSB5cxx11E[.refptr._ZN7knn_jni10PARAMETERSB5cxx11E]+0x0): undefined reference to `knn_jni::PARAMETERS[abi:cxx11]'
collect2.exe: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/opensearchknn_nmslib.dir/build.make:120: release/opensearchknn_nmslib.dll] Error 1
make[2]: *** [CMakeFiles/Makefile2:287: CMakeFiles/opensearchknn_nmslib.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:294: CMakeFiles/opensearchknn_nmslib.dir/rule] Error 2
make: *** [Makefile:202: opensearchknn_nmslib] Error 2

> Task :buildJniLib FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildJniLib'.
> Process 'command 'make'' finished with non-zero exit value 2

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to ht
```

--- a/build.gradle
+++ b/build.gradle
@@ -313,7 +313,7 @@ task cmakeJniLib(type:Exec) {
 task buildJniLib(type:Exec) {
     dependsOn cmakeJniLib
     workingDir 'jni'
-    commandLine 'make', 'opensearchknn_nmslib', 'opensearchknn_faiss'
+    commandLine 'make', 'opensearchknn_nmslib', 'opensearchknn_faiss', 'opensearchknn_common'
 }
 
 test {

--- a/jni/CMakeLists.txt
+++ b/jni/CMakeLists.txt
@@ -9,7 +9,9 @@ project(KNNPlugin_JNI)
 
 # ---------------------------------- SETUP ----------------------------------
 # Target libraries to be compiled
-set(TARGET_LIB_COMMON opensearchknn_common)  # Shared library with common utilities
+# Shared library with common utilities. Not a JNI library. Other JNI libs should depend on this one.
+set(TARGET_LIB_UTIL opensearchknn_util)
+set(TARGET_LIB_COMMON opensearchknn_common)  # common lib for JNI
 set(TARGET_LIB_NMSLIB opensearchknn_nmslib)  # nmslib JNI
 set(TARGET_LIB_FAISS opensearchknn_faiss)    # faiss JNI
 set(TARGET_LIBS "")  # Libs to be installed
@@ -60,8 +62,25 @@ elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL x86_64)
 endif()
 # ----------------------------------------------------------------------------
 
+# ---------------------------------- UTIL ----------------------------------
+add_library(${TARGET_LIB_UTIL} SHARED ${CMAKE_CURRENT_SOURCE_DIR}/src/jni_util.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/commons.cpp)
+target_include_directories(${TARGET_LIB_UTIL} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include $ENV{JAVA_HOME}/include $ENV{JAVA_HOME}/include/${JVM_OS_TYPE})
+set_target_properties(${TARGET_LIB_UTIL} PROPERTIES SUFFIX ${LIB_EXT})
+set_target_properties(${TARGET_LIB_UTIL} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+if (NOT "${WIN32}" STREQUAL "")
+    # Use RUNTIME_OUTPUT_DIRECTORY, to build the target library (opensearchknn_util) in the specified directory at runtime.
+    set_target_properties(${TARGET_LIB_UTIL} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/release)
+else()
+    set_target_properties(${TARGET_LIB_UTIL} PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/release)
+endif()
+
+list(APPEND TARGET_LIBS ${TARGET_LIB_UTIL})
+# ----------------------------------------------------------------------------
+
 # ---------------------------------- COMMON ----------------------------------
-add_library(${TARGET_LIB_COMMON} SHARED ${CMAKE_CURRENT_SOURCE_DIR}/src/jni_util.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/org_opensearch_knn_jni_JNICommons.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/commons.cpp)
+add_library(${TARGET_LIB_COMMON} SHARED ${CMAKE_CURRENT_SOURCE_DIR}/src/org_opensearch_knn_jni_JNICommons.cpp)
+target_link_libraries(${TARGET_LIB_COMMON} ${TARGET_LIB_UTIL})
 target_include_directories(${TARGET_LIB_COMMON} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include $ENV{JAVA_HOME}/include $ENV{JAVA_HOME}/include/${JVM_OS_TYPE})
 set_target_properties(${TARGET_LIB_COMMON} PROPERTIES SUFFIX ${LIB_EXT})
 set_target_properties(${TARGET_LIB_COMMON} PROPERTIES POSITION_INDEPENDENT_CODE ON)
@@ -103,7 +122,7 @@ if (${CONFIG_NMSLIB} STREQUAL ON OR ${CONFIG_ALL} STREQUAL ON OR ${CONFIG_TEST} 
     add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/nmslib/similarity_search)
 
     add_library(${TARGET_LIB_NMSLIB} SHARED ${CMAKE_CURRENT_SOURCE_DIR}/src/org_opensearch_knn_jni_NmslibService.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/nmslib_wrapper.cpp)
-    target_link_libraries(${TARGET_LIB_NMSLIB} NonMetricSpaceLib ${TARGET_LIB_COMMON})
+    target_link_libraries(${TARGET_LIB_NMSLIB} NonMetricSpaceLib ${TARGET_LIB_UTIL})
     target_include_directories(${TARGET_LIB_NMSLIB} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include $ENV{JAVA_HOME}/include $ENV{JAVA_HOME}/include/${JVM_OS_TYPE} ${CMAKE_CURRENT_SOURCE_DIR}/external/nmslib/similarity_search/include)
     set_target_properties(${TARGET_LIB_NMSLIB} PROPERTIES SUFFIX ${LIB_EXT})
     set_target_properties(${TARGET_LIB_NMSLIB} PROPERTIES POSITION_INDEPENDENT_CODE ON)
@@ -190,7 +209,7 @@ if (${CONFIG_FAISS} STREQUAL ON OR ${CONFIG_ALL} STREQUAL ON OR ${CONFIG_TEST} S
         ${CMAKE_CURRENT_SOURCE_DIR}/src/faiss_wrapper.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/src/faiss_util.cpp
     )
-    target_link_libraries(${TARGET_LIB_FAISS} ${TARGET_LINK_FAISS_LIB} ${TARGET_LIB_COMMON} OpenMP::OpenMP_CXX)
+    target_link_libraries(${TARGET_LIB_FAISS} ${TARGET_LINK_FAISS_LIB} ${TARGET_LIB_UTIL} OpenMP::OpenMP_CXX)
     target_include_directories(${TARGET_LIB_FAISS} PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/include
         $ENV{JAVA_HOME}/include
@@ -249,6 +268,7 @@ if ("${WIN32}" STREQUAL "")
                 ${TARGET_LIB_FAISS}
                 ${TARGET_LIB_NMSLIB}
                 ${TARGET_LIB_COMMON}
+                ${TARGET_LIB_UTIL}
         )
 
         target_include_directories(jni_test PRIVATE
@@ -266,55 +286,4 @@ if ("${WIN32}" STREQUAL "")
     endif ()
 endif()
 
-# ---------------------------------------------------------------------------
-
-# -------------------------------- INSTALL ----------------------------------
-# Installation rules for shared library
-install(TARGETS ${TARGET_LIBS}
-        LIBRARY DESTINATION lib
-        COMPONENT library)
-
-set(KNN_MAINTAINER "OpenSearch Team <opensearch@amazon.com>")
-set(OPENSEARCH_DOWNLOAD_URL "https://opensearch.org/downloads.html")
-set(CPACK_PACKAGE_NAME ${KNN_PACKAGE_NAME})
-set(CPACK_PACKAGE_VERSION ${KNN_PLUGIN_VERSION})
-set(CMAKE_INSTALL_PREFIX /usr)
-set(CPACK_GENERATOR "RPM;DEB")
-set(CPACK_OUTPUT_FILE_PREFIX packages)
-set(CPACK_PACKAGE_RELEASE 1)
-set(CPACK_PACKAGE_VENDOR "Amazon")
-set(CPACK_PACKAGE_CONTACT "Maintainer: ${KNN_MAINTAINER}")
-set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
-set(CPACK_COMPONENTS_GROUPING IGNORE)
-get_cmake_property(CPACK_COMPONENTS_ALL COMPONENTS)
-list(REMOVE_ITEM CPACK_COMPONENTS_ALL "Unspecified")
-
-# Component variable
-set(KNN_PACKAGE_NAME opensearch-knnlib)
-set(KNN_PACKAGE_DESCRIPTION "KNN JNI libraries built off of nmslib and faiss for OpenSearch")
-
-# RPM
-set(CPACK_RPM_PACKAGE_LICENSE "ASL-2.0")
-set(CPACK_RPM_COMPONENT_INSTALL ON)
-set(CPACK_RPM_PACKAGE_URL ${OPENSEARCH_DOWNLOAD_URL})
-set(CPACK_RPM_PACKAGE_RELEASE ${CPACK_PACKAGE_RELEASE})
-
-set(CPACK_RPM_PACKAGE_NAME ${KNN_PACKAGE_NAME})
-set(CPACK_RPM_FILE_NAME "${CPACK_RPM_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${JVM_OS_TYPE}-${MACH_ARCH}.rpm")
-set(CPACK_RPM_PACKAGE_DESCRIPTION ${KNN_PACKAGE_DESCRIPTION})
-set(CPACK_RPM_PACKAGE_SUMMARY "OpenSearch k-NN JNI Library with nmslib and faiss")
-
-# DEB
-set(CPACK_DEBIAN_PACKAGE_HOMEPAGE ${OPENSEARCH_DOWNLOAD_URL})
-set(CPACK_DEBIAN_PACKAGE_MAINTAINER ${KNN_MAINTAINER})
-set(CPACK_DEBIAN_PACKAGE_VERSION ${CPACK_PACKAGE_VERSION})
-set(CPACK_DEBIAN_PACKAGE_SECTION "libs")
-set(CPACK_DEB_COMPONENT_INSTALL ON)
-
-set(CPACK_DEBIAN_PACKAGE_NAME ${KNN_PACKAGE_NAME})
-set(CPACK_DEBIAN_FILE_NAME "${CPACK_DEBIAN_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${JVM_OS_TYPE}-${MACH_ARCH}.deb")
-set(CPACK_DEBIAN_DESCRIPTION ${KNN_PACKAGE_DESCRIPTION})
-set(CPACK_DEBIAN_PACKAGE_SOURCE ${CPACK_DEBIAN_PACKAGE_NAME})
-
-include(CPack)
 # ---------------------------------------------------------------------------

